src/Catalog/Dnx/DnxCatalogCollector.cs

```diff
@@ -173,6 +195,70 @@ private async Task UpdatePackageVersionIndexAsync(IEnumerable<CatalogEntry> cata
             string version,
             Uri sourceUri,
             CancellationToken cancellationToken)
+        {
+            var shouldUseSourceStorage = _sourceStorage != null;
+            var sourceBlob = await GetBlockBlobAsync(id, version) as AzureCloudBlockBlob;
+
+            if (sourceBlob != null)
+            {
+                await sourceBlob.FetchAttributesAsync(cancellationToken);
+
+                // It's possible (though unlikely) that the blob may change between reads.  Reading a blob with a
+                // single GET request returns the whole blob in a consistent state, but we're reading the blob many
+                // different times.  To detect the blob changing between reads, we check the ETag again later.
+                // If the ETag's differ, we'll fall back to using a single HTTP GET request.
+                var etag = sourceBlob.ETag;
+
+                string nuspec = await GetNuspecAsync(sourceBlob, id, cancellationToken);
+
+                if (string.IsNullOrEmpty(nuspec))
+                {
+                    _logger.LogWarning("No .nuspec available for {Id}/{Version}. Skipping.", id, version);
```

Comment:
>Skipping [](start = 81, length = 8)

Is `Skipping` correct?  It looks like this package will be processed by `ProcessPackageDetailsViaHttpAsync`